tests/utils/test_utils.py

```diff
@@ -352,6 +368,13 @@ def test_check_timeseries_input():
     assert n_voxels == 1
     assert n_subjects == 10
 
+    # Check that 4-dimensional input array throws error
+    array_4d = array_3d[..., np.newaxis]
+    try:
+        (data_array_4d, _, _, _) = _check_timeseries_input(array_4d)
+    except ValueError:
+        pass
```

Comment:
Pytest has a nice way to check exceptions:
https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Up to you.

Reply:
Cool! Incorporated this and looks a bit cleaner.